src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs

```diff
@@ -274,6 +274,38 @@ public static string GetTempPath(string workingDir)
             }
         }
 
+        private static string FindDotNetExecutable()
+        {
+            var expectedPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH");
+            if (!string.IsNullOrEmpty(expectedPath))
+            {
+                return expectedPath;
+            }
+
+#if NET
+            expectedPath = System.Environment.ProcessPath;
+#else
+            expectedPath = Process.GetCurrentProcess().MainModule.FileName;
+#endif
+
+            if (!Path.GetFileNameWithoutExtension(expectedPath).Equals("dotnet"))
```

Comment:
Nit: technically safer if `Path.GetFileNameWithoutExtension` returned `null`.

```suggestion
            if (Path.GetFileNameWithoutExtension(expectedPath) != "dotnet")
```

Reply:
I don't think that's a risk here since the currently executing process should have a path, but I can make this a yoda condition anyway.